### PR TITLE
PR Review Process and Submission Guidelines

### PR DESCRIPTION
**Pull request by kantai**
This is a pretty skeletal CONTRIBUTING doc that roughly describes the PR review process, and what the PR author and PR reviewer are responsible for.

The goal here is to highlight what part of the review process affects QA specifically for Gaia.

The important things to check for QA in Gaia are:

1. Whether or not authentication tokens are affected by a particular PR.
2. Whether or not the PR changes how the deployed hub writes or expects to read data.
3. Whether or not the PR expects to add significant load to the hub server.

And so this was added to the explicit instructions for a reviewer.

### CODE REVIEW

**Comment by kantai** [comment]
Repinging @zone117x and @jcnelson for a review here!

**Comment by jcnelson** [comment]
:+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+2. Describe how that goal was achieved through the submitted implementation.
+3. The code must obey our eslint definitions, pass our unit tests, and
+   contain correct TypeFlow annotations.
+4. Contain tests that cover any and all new functionality or code changes.
+5. Describe how the new functionality can be tested manually.
+6. Document any new features or endpoints, and describe how developers
+   would be expected to interact with them.
+7. PR authors should agree to our contributor's agreement.
+
+PRs on Gaia should be reviewed by at least (2) maintainers.
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: @kantai Do we want to set some squash and merge guidelines. Many projects require outside contributors to submit changes as a single commit (squash and merge). This allows for a quick rollback when problems arise.

Comment (kantai) [reply CONTRIBUTING.md]: Yes -- though our preference is to do "non-squashing" merges. I'd rather see the whole history (apart from obvious commit spam).

**Pull request merged**